### PR TITLE
Better reporting of "unexpected node" error messages

### PR DESCRIPTION
**Pull request by lucteo**
Previously, the code would report things like `unexpected class member` without telling the user what node kind it found and at which location in the Val source file.

With this change we improve the error reporting to:
```
.../CXXTranspiler.swift:114: Fatal error: unexpected class member, found FunctionDecl at location: .../PostfixOperators.val:12:3
```

### CODE REVIEW

**Review by kyouko-taiga — APPROVED** [review]
Nice!

**Comment by lucteo** [comment]
@kyouko-taiga : I merged the PR sitting on top of this, and it merged it into this branch. Sorry for extra changes in the PR.
I also fixed compilation after merging from main.

Probably the easiest way to re-review this is to look at the last commits.

File: Sources/CodeGen/CXX/CXXTranspiler.swift [hunk 1]
```diff
@@ -64,7 +64,7 @@ public struct CXXTranspiler {
         fatalError("not implemented")
 
       default:
-        unreachable()
+        unreachable("unexpected type")
```
Comment (kyouko-taiga) [inline Sources/CodeGen/CXX/CXXTranspiler.swift:67]: Don't you want to use your `unexpected` function here?

**Review by kyouko-taiga — APPROVED** [review]

File: Sources/CodeGen/CXX/CXXTranspiler.swift (hunk 1 above)
Comment (lucteo) [reply Sources/CodeGen/CXX/CXXTranspiler.swift:67]: My abstraction doesn't work for types (don't have the location).

I can implement an abstraction, but I would do this in a different PR.

Comment (lucteo) [reply Sources/CodeGen/CXX/CXXTranspiler.swift:67]: Created #335

**Pull request merged**